include/seqan3/io/sequence/sequence_file_format_fasta.hpp

```diff
@@ -242,56 +221,49 @@ class sequence_file_format_fasta
-                                   {
-                                       if (!is_legal_alph(c))
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+        {
+            is_in_alphabet<seq_legal_alph_type> const is_legal_alph;
+            ranges::copy(stream_view | ranges::view::take_while(!is_id)             // until next header (or end)
+                                     | ranges::view::remove_if(is_space || is_digit)// ignore whitespace and numbers
+                                     | view::transform([is_legal_alph] (char const c)
                                        {
-                                           throw parse_error{std::string{"Encountered an unexpected letter: "} +
-                                                             is_legal_alph.msg.string() +
-                                                             " evaluated to false on " +
-                                                             detail::make_printable(c)};
-                                       }
-                                       return c;
-                                   })                                              // enforce legal alphabet
-                                 | view::char_to<value_type_t<seq_type>>,          // convert to actual target alphabet
-                     detail::make_conversion_output_iterator(seq));
-    }
-
-    //!\brief Implementation of reading the sequence; overload for std::ignore (should be slightly faster).
-    template <typename      stream_view_t,
-              typename      seq_legal_alph_type>
-    void read_seq(stream_view_t                                       & stream_view,
-                  sequence_file_in_options<seq_legal_alph_type> const &,
-                  detail::ignore_t                              const &)
-    {
-        auto seq_view = stream_view | ranges::view::take_while(!is_char<'>'>{});     // until next header (or end)
-
-        for (auto it = ranges::begin(seq_view); it != ranges::end(seq_view); ++it)
-        {}
+                                           if (!is_legal_alph(c))
+                                           {
+                                               throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                   is_legal_alph.msg.string() +
+                                                                   " evaluated to false on " +
+                                                                   detail::make_printable(c)};
+                                           }
+                                           return c;
+                                       })                                           // enforce legal alphabet
+                                     | view::char_to<value_type_t<seq_type>>,       // convert to actual target alphabet
+                         detail::make_conversion_output_iterator(seq));
```

Comment:
It's not clear to me: why do we need `detail::make_conversion_output_iterator` instead of `ranges::back_inserter` here?

Reply:
In general the detail::make_conversion_output_iterator is deprecated and should be removed by @rrahn. The reason I am still using it in some places, is that we sometimes need an custom overload if the parameter is actually `std::ignore`. I guess we should should make our own back_insert_iterator that inherits from `ranges::back_insert_iterator` by default, but is specialised for `std::ignore`. Do you want to make a PR? :smiley: